draco/analysis/delay.py

```diff
@@ -846,8 +889,56 @@ def fourier_matrix_c2r(N, fsel=None):
     return Fr
 
 
+def fourier_matrix_c2c(N, fsel=None):
+    """Generate a Fourier matrix to represent a complex to complex FFT.
+
+    Parameters
+    ----------
+    N : integer
+        Length of timestream that we are transforming to.
+    fsel : array_like, optional
+        Indices of the frequency channels to include in the transformation
+        matrix. By default, assume all channels.
+
+    Returns
+    -------
+    F : np.ndarray
+        An array performing the Fourier transform from a complex time series to
```

Comment:
This seems subtle, and certainly isn't true as written.

I can take an arbitrary complex time series, apply this matrix to it, and I will get the usual *complex* Fourier transform frequencies inserted into the `[0::2]` elements of the output vector, and `1j` times them in the `[1::2]` elements.

If the matrix here was being used to define the transpose (i.e. mapping from real-valued, real and imaginary frequency components to a complex time stream) this would be correct, but it's not.

I really think that the elements in `F[0::2]` are all thats needed to represent this, but it would require changes in the way that `delay_spectrum_gibbs` is implemented. Alternatively, you could have this mapping from real-imag alternating floats on both sides of the matrix (which is probably what you'd get from `F.view(np.float64)`).

Reply:
> If the matrix here was being used to define the transpose (i.e. mapping from real-valued, real and imaginary frequency components to a complex time stream) this would be correct, but it's not.

I wrote this routine such that the transpose of its output is correct, since we only ever use the transpose. (It always struck me as strange that `fourier_matrix_c2r()` was defined in such a way that its transpose needs to be taken to get the result we want, but I wrote `fourier_matrix_c2c()` to match this convention.) I remember checking the outputs against the numpy ifft routine (which motivated the comments that describe the comparison), so I believe the transpose of `fourier_matrix_c2c()` is correct.

Reply:
The rationale is that as it's a classic inverse problem, you first define the forward model, which is given by F. You do kind of use it within the $\mathbf{F}^\dagger \mathbf{N}^{-1} \mathbf{F}$ term.

I'll try and think about it all today, and get a coherent picture in my head. I think correctly expressing the forward model in the way we've written requires using a matrix like
```math
\begin{pmatrix}
\mathbf{F} / 2 & \mathbf{F}^* / 2 \\
-i \mathbf{F} / 2 & -i \mathbf{F}^* / 2
\end{pmatrix}
\begin{pmatrix}
\mathbf{t} \\
\mathbf{t}^*
\end{pmatrix}
```
with $\mathbf{F}_{ab} = e^{2 \pi i a b / N}$, which forces you back into the full statistics of complex Gaussian numbers (i.e. needing both correlation and relation matrices).

Reply:
Hmmm.... I think Github's new math rendering is meant to have rendered the block above, but it's only done the inline parts. Not quite sure what's going on with that.

Reply:
Made it work 🥳